src/org/zaproxy/zap/extension/ascanrulesAlpha/GetForPostScanner.java

```diff
@@ -0,0 +1,148 @@
+import org.parosproxy.paros.core.scanner.Category;
+import org.parosproxy.paros.network.HtmlParameter;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpRequestHeader;
+
+/**
+ * Active scan rule which checks whether or not POST requests with parameters
+ * are accepted as GET equivalent requests.
+ * 
+ * @author kingthorin+owaspzap@gmail.com
+ */
+public class GetForPostScanner extends AbstractAppPlugin {
+
+	private static final Logger LOG = Logger.getLogger(GetForPostScanner.class);
+	private static final String MESSAGE_PREFIX = "ascanalpha.getforpostscanner.";
+	private static final int PLUGIN_ID = 10058; 
+
+	@Override
+	public int getId() {
+		return PLUGIN_ID;
+	}
+
+	@Override
+	public String getName() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "name");
+	}
+
+	@Override
+	public String getDescription() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "desc");
+	}
+
+	@Override
+	public int getCategory() {
+		return Category.MISC;
+	}
+
+	@Override
+	public String getSolution() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "soln");
+	}
+	
+	@Override
+	public String getReference() {
+		return null;
+	}
+
+	@Override
+	public void init() {
+
+	}
+
+	@Override
+	public void scan() {
+		// Check if the user stopped things. One request per URL so check before
+		// sending the request
+		if (isStop()) {
+			if (LOG.isDebugEnabled()) {
+				LOG.debug("Scanner " + getName() + " Stopping.");
+			}
+			return;
+		}
+		
+		HttpMessage baseMsg = getBaseMsg();
+		TreeSet<HtmlParameter> postParams = baseMsg.getFormParams();
+		if (!baseMsg.getRequestHeader().getMethod().equals(HttpRequestHeader.POST)
+				|| postParams.first().getName().isEmpty()) {
+			return; //Not a POST or no form params, no reason to continue
+		}
+		
+		HttpMessage newRequest = getNewMsg();
+		newRequest.getRequestHeader().setMethod(HttpRequestHeader.GET);
+		newRequest.setFormParams(new TreeSet<HtmlParameter>());
+		for (HtmlParameter param : postParams) {
+			param.setType(HtmlParameter.Type.url);
+		}
+		newRequest.getRequestHeader().setGetParams(postParams);
+		
+		try {
+			sendAndReceive(newRequest);
+		} catch (IOException e) {
+			LOG.warn("An error occurred while checking [" + newRequest.getRequestHeader().getMethod() + "] ["
+					+ newRequest.getRequestHeader().getURI().toString() + "] for " + getName() + " Caught "
+					+ e.getClass().getName() + " " + e.getMessage());
+		}
+		
+		if (newRequest.getResponseBody().equals(baseMsg.getResponseBody())) {
+			bingo(getRisk(), // Risk
+					Alert.CONFIDENCE_HIGH, // Confidence
+					getName(), // Name
+					getDescription(), // Description
+					baseMsg.getRequestHeader().getURI().toString(), // URI
+					null, // Param
+					"", // Attack
+					"", // OtherInfo
+					getSolution(), // Solution
+					newRequest.getRequestHeader().getPrimeHeader(), // Evidence
+					getCweId(), // CWE ID
+					getWascId(), // WASC ID
+					baseMsg); // HTTPMessage
```

Comment:
It should use the `newRequest` message, which is the one that contains the evidence.